### PR TITLE
replace validate_* methods with datatypes

### PR DESCRIPTION
**Pull request by bastelfreak**
The goal of this is to get rid of most of the validate_* function calls since they are now deprecated.

### CODE REVIEW

**Comment by mmoll** [comment]
this needs a rebase :)

**Comment by bastelfreak** [comment]
Why is this still open :(

File: metadata.json [hunk 1]
```diff
@@ -67,26 +67,26 @@
     "requirements": [
     {
       "name": "puppet",
-      "version_requirement": ">= 4.7.0"
+      "version_requirement": ">= 4.7.1 < 5.0.0"
```
Comment (eputnam) [inline metadata.json]: bump to < 6.0.0 plz

Comment (eputnam) [reply metadata.json]: just kidding

Comment (bastelfreak) [reply metadata.json]: tested it locally, doesnt work, reverted the commit. Fixing the issues for puppet5 should be handled in another PR.

**Pull request merged**